### DISCUSSION TITLE
How to retrieve current connected state?

### DISCUSSION THREAD

**Discussion by G4PLS** [aatikturk/obsws-python#68] [Q&A] [answered]
I was looking at the Code and was wondering if there is any way to retrieve the current state of the connection?
You can disconnect but I want to be able to retrieve when the connection closes from the OBS side. For example when OBS gets closed or when you kick a websocket session

**Comment by onyx-and-iris** [comment, marked as answer]
Hello, there exists an [ExitStarted](https://github.com/obsproject/obs-websocket/blob/master/docs/generated/protocol.md#exitstarted) event which fires when OBS is being shut down. You may refer to the [events example](https://github.com/aatikturk/obsws-python/blob/main/examples/events/__main__.py) for usage. 

As for kicking a websocket session well it seems the ReqClient will throw a `JSONDecodeError` on the next request attempt, which admittedly could be wrapped in a more useful exception. As for the EventClient, it will just hang since recv() is blocking and simply waiting for the next event. I would suggest you rely on the ExitStarted event if you need your client program to be notified of a shutdown.

I hope that's helpful.

**Comment by G4PLS** [comment]
This is very helpfull!
This was for me the only downside over obs-websocket-py but handling it with the Exit event would fix that easily.
